lib/kast.js

```diff
@@ -67,6 +67,56 @@ const joinMatches = (match1, match2) => {
 const antimatch = (pattern, term, path = []) => {
   if (pattern.node == KVARIABLE) {
     return [];
+  } else if (pattern.node == KAPPLY && term.node == KAPPLY && pattern.label == term.label && pattern.label == "_Map_") {
+    // gather keys lhs
+    const lhs = pattern.args
+      .filter(pair => pair.node !== "KVariable")
+      .map(pair => [format(pair.args[0]), pair.args[1]])
+    const rhs = term.args
+      .filter(pair => pair.node !== "KVariable")
+      .map(pair => [format(pair.args[0]), pair.args[1]])
+    const lhs_keys = lhs.map(pair => pair[0]);
+    const rhs_keys = rhs.map(pair => pair[0]);
+    // TODO - introduce a type arg which states if arg is missing or rejected
+    const missing_keys = rhs
+      .filter(([key, v]) => !lhs_keys.includes(key))
+    const rejected_keys = lhs
+      .filter(([key, v]) => !rhs_keys.includes(key))
+    const matching_keys = lhs
+      .filter(([key, v]) => rhs_keys.includes(key))
+
+    const matching = matching_keys
+      .map(([key, pattern_]) => antimatch(pattern_, rhs.find(el => el[0] == key)[1], path.concat( key.replace(/\n\s*/g, "") )))
```

Comment:
I think a line like this one is a bit unreadable unless someone already has spent quite some time looking at this file.

As above, I think we should start improving the readability of klab by e.g. relying on variable names rather than the position in an array, and relying less on one-liners.

E.g. the function passed to `find` could be made multiline and declare some vars.

My 2c and this function is just an example, the same applies to many other places in klab.

Reply:
:( i like one liners

Reply:
:( i like one liners